### PR TITLE
Bring the blog into the main app

### LINKED ISSUE #2860: Move the blog under the main SPI site
I've been wanting to do this for a long time, as the local development/preview environment is so out of date I can't actually preview posts locally at the moment. It was never high enough priority until now, though, as this also contributes towards the work we are doing in https://github.com/orgs/SwiftPackageIndex/projects/3.

This will be a boost for our SEO as it brings all these words in under the main domain. We also get a slightly modernised blog with a consistent header/footer and the ability to easily bring in things like the podcast CTA and supporters really easily. The old blog didn't have an index page either. It was time!

We will redirect all old blog URLs and of course the feed URL.

### Go-Live Plan

- [x] Set up permanent redirects in a static site hosted by Netlify¹
- [x] Deploy the new blog to staging and test everything
- [x] Deploy the new blog to production with the old blog still live
- [x] Test all redirects, including the RSS feed redirect
- [x] Change the DNS records to point at Netlify

### Post-Live Tasks

- [x] Fix up all internal SPI blog links so they *don’t* go through redirects
- [x] Open a PR against Swift.org fixing up blog post links without a redirect
- [x] Archive https://github.com/SwiftPackageIndex/SwiftPackageIndex-Blog
- [x] Delete https://github.com/SwiftPackageIndex/SwiftPackageIndex.github.io

¹ This can't be hosted on GitHub pages as they do not support redirects.

### PR DESCRIPTION
**Pull request by daveverwer**
Progresses #2860 

Not yet ready for review, but close enough that I need a PR open.

### Things still to do

- [x] Add RSS autodiscovery tags to the homepage and the blog pages
- [x] Snapshot test for the blog index page
- [x] Snapshot test for a sample blog page

### CODE REVIEW

File: Sources/App/Views/PublicPage.swift [hunk 1]
```diff
@@ -90,10 +91,21 @@ class PublicPage {
...
+    func preHead() -> Node<HTML.HeadContext> {
+        return .empty
+    }
+
+    /// Any additional tags that should be inserted as the *last* tags in the `<head>`.
+    func postHead() -> Node<HTML.HeadContext> {
+        return .empty
+    }
+
     /// if a search engine requests the page, we can tell it not to index it by including this meta tag.
```
Comment (daveverwer) [inline Sources/App/Views/PublicPage.swift:109]: These ended up not being used, but they are useful overloads into the `head` so I'd like to keep them.

File: Sources/App/Views/Blog/BlogActions+Model.swift [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    // Only "published" posts show in production.
+                    allSummaries.filter { $0.published }
+                } else {
+                    // Show *everything* in development and on staging.
+                    allSummaries
+                }
+            } catch {
+                Current.logger().report(error: error)
+                summaries = []
+            }
```
Comment (finestructure) [inline Sources/App/Views/Blog/BlogActions+Model.swift:56]: We should either rethrow the exception or _not_ mark `init` as `throws`. Right now it's `throws` but doesn't need to be.

File: Sources/App/Views/Blog/BlogActions+Model.swift [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            var slug: String
+            var title: String
+            var summary: String
+            var publishedAt: Date
+            var published: Bool
+        }
+
+        var summaries: [PostSummary]
+
+        init() throws {
```
Comment (daveverwer) [inline Sources/App/Views/Blog/BlogActions+Model.swift:34]:
```suggestion
        init() {
```

**Review by finestructure — COMMENTED** [review]
I'll open a PR with a couple of test proposed test tweaks

**Comment by daveverwer** [comment]
Re: https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/pull/2861#discussion_r1466455725

I noted you didn't change this in your PR. Do you feel strongly about it?

**Review by finestructure — APPROVED** [review]

**Comment by daveverwer** [comment]
Deploying as 2.98.5

**Pull request merged**